pygitguardian/client.py

```diff
@@ -584,7 +584,10 @@ def read_metadata(self) -> Optional[Detail]:
             result.status_code = resp.status_code
             return result
         metadata = ServerMetadata.from_dict(resp.json())
-
+        if "general__maximum_payload_size" in metadata.preferences:
+            metadata.secret_scan_preferences.general__maximum_payload_size = (
+                metadata.preferences["general__maximum_payload_size"]
+            )
```

Comment:
I think the maximum payload size should be a property of GGClient: all verticals could need it. I would turn this into:

```python
self.maximum_payload_size = metadata.preferences.get("general__maximum_payload_size", MAXIMUM_PAYLOAD_SIZE)
```